cleanrl/sac_atari.py

```diff
@@ -0,0 +1,359 @@
+        env = gym.wrappers.ResizeObservation(env, (84, 84))
+        env = gym.wrappers.GrayScaleObservation(env)
+        env = gym.wrappers.FrameStack(env, 4)
+        env.seed(seed)
+        env.action_space.seed(seed)
+        env.observation_space.seed(seed)
+        return env
+
+    return thunk
+
+
+# He initialization
+def network_init(net, bias_const=0.0):
+    for param in net.parameters():
+        if type(param) == nn.Linear or type(param) == nn.Conv2d:
+            nn.init.kaiming_normal_(param.weight)
+            torch.nn.init.constant_(param.bias, bias_const)
+
+    return net
+
+
+# ALGO LOGIC: initialize agent here:
+# NOTE: Sharing a CNN encoder between Actor and Critics is not recommended for SAC
+# See https://arxiv.org/abs/1910.01741 for more info
+# TL;DR The actor's gradients mess up the representation when using a joint encoder
+class SoftQNetwork(nn.Module):
+    def __init__(self, envs):
+        super().__init__()
+        obs_shape = envs.single_observation_space.shape
+        self.conv = nn.Sequential(
+            nn.Conv2d(obs_shape[0], 32, kernel_size=8, stride=4),
+            nn.ReLU(),
+            nn.Conv2d(32, 64, kernel_size=4, stride=2),
+            nn.ReLU(),
+            nn.Conv2d(64, 64, kernel_size=3, stride=1),
+            nn.Flatten(),
+            nn.ReLU(),
+        )
+
+        with torch.inference_mode():
+            output_dim = self.conv(torch.zeros(1, *obs_shape)).shape[1]
+
+        self.Q_func = nn.Sequential(
+            nn.Linear(output_dim, 512),
+            nn.ReLU(),
+            nn.Linear(512, envs.single_action_space.n),
+        )
+
+    def forward(self, x):
+        x = self.conv(x / 255.0)
+        x = self.Q_func(x)
+        return x
+
+
+class Actor(nn.Module):
+    def __init__(self, envs):
+        super().__init__()
+        obs_shape = envs.single_observation_space.shape
+        self.conv = nn.Sequential(
+            nn.Conv2d(obs_shape[0], 32, kernel_size=8, stride=4),
+            nn.ReLU(),
+            nn.Conv2d(32, 64, kernel_size=4, stride=2),
+            nn.ReLU(),
+            nn.Conv2d(64, 64, kernel_size=3, stride=1),
+            nn.Flatten(),
+            nn.ReLU(),
+        )
+
+        with torch.inference_mode():
+            output_dim = self.conv(torch.zeros(1, *obs_shape)).shape[1]
+
+        self.fc_logits = nn.Sequential(
+            nn.Linear(output_dim, 512),
+            nn.ReLU(),
+            nn.Linear(512, envs.single_action_space.n),
+        )
+
+    def forward(self, x):
+        x = self.conv(x)
+        logits = self.fc_logits(x)
+
+        return logits
+
+    def get_action(self, x):
+        logits = self(x / 255.0)
+        policy_dist = Categorical(logits=logits)
+        action = policy_dist.sample()
+        # Action probabilities for calculating the adapted soft-Q loss
+        action_probs = policy_dist.probs
+        log_pi = F.log_softmax(action_probs, dim=1)
+        return action, log_pi, action_probs
```

Comment:
Could we use `log_prob` instead, so that the naming are standardized across continuous and discrete action space tasks ?
https://github.com/vwxyzjn/cleanrl/blob/e466f6efb251462de2b80f064133f32cb5d83e22/cleanrl/sac_continuous_action.py#L143
Thanks.